### PR TITLE
Run only tests from modified pipelines

### LINKED ISSUE #20: setup CI: run only tests from modified pipelines
Improve all the workflows so only tests for pipelines modified in current PR are run.

**Motivation**
1. My biggest worry is that our API Credentials and accounts will get locked due to excessive CI traffic.
2. People should be able to merge their changes even if other pipelines are (temporarily) out of order.

**Tasks**
1. * [ ] find the affected files by comparing current branch to master
2. * [ ] identify pipelines and run tests only for them
3. * [ ] apply to all workflows

### PR DESCRIPTION
**Pull request by burnash**
Implements conditional testing depending on modifications in `pipelines` directory.

Closes #20

### CODE REVIEW

**Comment by burnash** [comment]
@rudolfix I extracted the check for changes in pipelines into a reusable workflow and included it in both `test_destinations.yml` and `test_destinations_slow.yml`. 

I left the lint workflow intact because it calls `poetry run pytest tests/test_dlt_init.py` in the lint target, which in turn calls `test_init_list_pipelines`. This function iterates over the pipelines folder. Let me know if you would like me to scope the linters, and I will update them in the Makefile.

File: .github/workflows/get_changed_pipelines.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Get changed pipelines
+        id: get_changed_pipelines
+        run: |
+          echo "base.sha: ${{ github.event.pull_request.base.sha }}"
```
[the remark this answers is not in the document]
Comment (rudolfix) [reply .github/workflows/get_changed_pipelines.yml:21]: windows support could be ignored

**Review by rudolfix — APPROVED** [review]
it looks good. please announce this change on our dlt-contributors or community updates. let's do https://github.com/dlt-hub/dlt/issues/213 on `dlt` repo

File: .github/workflows/get_changed_pipelines.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (burnash) [reply .github/workflows/get_changed_pipelines.yml:21]: Yes, we'd need to adjust it since the SHAs are provided by GH Actions.

**Pull request merged**